SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -426,26 +426,31 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	}
 #endif
 
-	if (IsServer() && SpatialSettings->bEnableUnrealLoadBalancer)
+	if (SpatialSettings->bEnableUnrealLoadBalancer)
 	{
-		if (SpatialSettings->LoadBalanceStrategy == nullptr)
-		{
-			UE_LOG(LogSpatialOSNetDriver, Error, TEXT("If EnableUnrealLoadBalancer is set, there must be a LoadBalancing strategy set. Using a 1x1 grid."));
-			LoadBalanceStrategy = NewObject<UGridBasedLBStrategy>(this);
-		}
-		else
-		{
-			// TODO: zoning - Move to AWorldSettings subclass [UNR-2386]
-			LoadBalanceStrategy = NewObject<UAbstractLBStrategy>(this, SpatialSettings->LoadBalanceStrategy);
+		if (IsServer()) {
+			if (SpatialSettings->LoadBalanceStrategy == nullptr)
+			{
+				UE_LOG(LogSpatialOSNetDriver, Error, TEXT("If EnableUnrealLoadBalancer is set, there must be a LoadBalancing strategy set. Using a 1x1 grid."));
+				LoadBalanceStrategy = NewObject<UGridBasedLBStrategy>(this);
+			}
+			else
+			{
+				// TODO: zoning - Move to AWorldSettings subclass [UNR-2386]
+				LoadBalanceStrategy = NewObject<UAbstractLBStrategy>(this, SpatialSettings->LoadBalanceStrategy);
+			}
+			LoadBalanceStrategy->Init(this);
 		}
-		LoadBalanceStrategy->Init(this);
 
 		VirtualWorkerTranslator = MakeUnique<SpatialVirtualWorkerTranslator>();
```

Comment:
I don't think we want the translator on the client.  Would prefer if just the relevant data was passed across to SpatialDebugger and replicated down to the client (this lets us disable the debugger in ship config, and not have core classes on the client that aren't needed)

Reply:
This is a lack of understanding on my part, my impression the SpatialDebugger logic ran purely on the client. Ultimately, the translator mapping component data is already available to the client via the GSM query that the client does on startup. For me, instantiating the VirtualWorkerTranslator on the client was just an easy way to parse this data and store locally. Conceptually, I agree replicating this information from the server-side is better and will be much easier to turn off from production. However, since we do already have the info arriving via the GSM so it was a smaller piece of work just to extra that way? A bigger reorg task seemed necessary to change the whole GSM querying to a much better server replicating entity data approach.

Reply:
If we are guaranteed that the client will see the data on the Translator component on the GSM via the QueryGSM() call then I think that's good as a first pass of this.  My longer term concern is (as you mention in a comment below) if the client doesn't have interest in the GSM, it won't get updates to the Physical/Virtual worker mapping - as it can change over time.

Reply:
I agree with @cmsmithio, we don't want the translator on the client and it's better to only pass what's necessary (Worker names in that case?).

Reply:
Imo that's best done with a minor change of how the SpatialDebugger works: https://improbableio.atlassian.net/browse/UNR-2565

Reply:
I also think it's unsafe to replicate something to client that client shouldn't be aware of, so I've added a comment to the ticket. I'm fine with resolving it, if @cmsmithio agrees.

Reply:
I think the approach Ally mentions in the ticket sounds good!  I agree about the security surface but we could disable the new debug actor component in shipping builds or similar - effectively making this a feature only developers have access to